app/views/shared/_header.html.haml

```diff
@@ -22,4 +22,4 @@
             %li
               %a(href="#") Account
               %a(href="#") Plan and Billing
-              = link_to 'Sign Out', sign_out_path
+              = link_to 'Sign Out', sign_out_path, method: :delete
```

Comment:
Never mind, I just pushed your app to Heroku and it works fine. There must be an outside factor on his app.